lib/wpc/msap.c

```diff
@@ -407,6 +408,43 @@ int msap_scratchpad_target_read_request(uint8_t * target_sequence_p,
     return result;
 }
 
+int msap_scratchpad_read_request(uint32_t start_address, uint8_t number_of_bytes, uint8_t * bytes)
+{
+    wpc_frame_t request, confirm;
+    int res;
+
+    if (number_of_bytes > MAXIMUM_SCRATCHPAD_BLOCK_SIZE)
+    {
+        LOGE("Block size is too big %d > %d\n", number_of_bytes, MAXIMUM_SCRATCHPAD_BLOCK_SIZE);
+        // Not necessary to send the request, directly return the mesh code
+        return 6;
+    }
+
+    LOGD("Read_request: start_address = %d, number_of_bytes = %d\n", start_address, number_of_bytes);
+    request.primitive_id = MSAP_SCRATCH_READ_REQUEST;
+    request.payload.msap_image_read_request_payload.start_add = start_address;
+    request.payload.msap_image_read_request_payload.number_of_bytes = number_of_bytes;
+    request.payload_length = sizeof(msap_image_read_req_pl_t);
+
+    res = WPC_Int_send_request_timeout(&request, &confirm, SCRATCHPAD_BLOK_TIMEOUT_MS);
```

Comment:
There is a typo in BLOK => BLOCK but not from your change
But feel free to fix it in this PR ;-)

Reply:
But I guess in case of reading, it should anyway be really fast to read so this 5s timeout is probably too big. In case of writing it may be longer but for reading the standard 500ms should be ok (maybe with very slow external flash)

Reply:
BLOK renamed to BLOCK, changed `WPC_Int_send_request_timeout()` to `WPC_Int_send_request()`.